lib/core/src/Cardano/Wallet.hs

```diff
@@ -369,7 +370,13 @@ newWalletLayer tracer block0 db nw tl = do
         -> s
         -> ExceptT ErrWalletAlreadyExists IO WalletId
     _createWallet wid wname s = do
-        let checkpoint = initWallet block0 s
+        let tmpTip = BlockHeader
+                (SlotId maxBound maxBound)
+                (Hash "<Hash of the block before genesis>")
+        let tmpCp = initWallet tmpTip s
```

Comment:
Hmmm... what? 
We should change `initWallet` to not take a block header but the actual first block. This will make the akward call to `applyBlocks` superfluous; initializing a wallet does the genesis block application (for what it's worth) and return an initial checkpoint.